### PR TITLE
[9.0][IMP] rma_operation_id on product and product.category

### PR DESCRIPTION
**Pull request by AaronHForgeFlow**
- This solves these issues: https://github.com/Eficent/stock-rma/issues/4 and https://github.com/Eficent/stock-rma/issues/5
- It also separates the rma operation menu between customer and supplier
- Added more improved tests
- Fixing existing previous travis errors
- Various fixes, especially in the wizard that creates supplier rma from a customer rma

### CODE REVIEW

**Comment by AaronHForgeFlow** [comment]
@lreficent This is ready to be merge :dancer:

File: rma/models/rma_order.py [hunk 1]
```diff
@@ -175,8 +179,8 @@ def action_rma_done(self):
     def _get_valid_lines(self):
         """:return: A recordset of rma lines.
         """
-        self.ensure_one()
-        return self.rma_line_ids
+        for rec in self:
+            return rec.rma_line_ids
```
Comment (LoisRForgeFlow) [inline rma/models/rma_order.py:183]: Why this? I prefer the approach with the `ensure_one` if you call this method with a recordset a error must raise. If not, you can be disregarding errors.

File: rma/models/rma_operation.py [hunk 2]
```diff
@@ -26,10 +26,7 @@ def _default_supplier_location_id(self):
 
     name = fields.Char('Description', required=True)
     code = fields.Char('Code', required=True)
-    refund_policy = fields.Selection([
```
Comment (LoisRForgeFlow) [inline rma/models/rma_operation.py:29]: Why is the refund policy removed?

Comment (LoisRForgeFlow) [reply rma/models/rma_operation.py:29]: Nevermind, I saw it is moved to rma_account

File: rma/models/product.py [hunk 3]
```diff
@@ -8,7 +8,9 @@
 class ProductTemplate(models.Model):
     _inherit = 'product.template'
 
-    rma_operation_id = fields.Many2one(
```
Comment (LoisRForgeFlow) [inline rma/models/product.py:11]: In the module 'rma_sale'  there are still some references to this field.

File: rma_purchase/wizards/rma_add_purchase.py [hunk 4]
```diff
@@ -75,8 +75,8 @@ def _prepare_rma_line_from_po_line(self, line):
             {'in_route_id': operation.in_route_id.id or route,
              'out_route_id': operation.out_route_id.id or route,
              'receipt_policy': operation.receipt_policy,
-             'location_id': operation.location_id.id or \
-                            self.env.ref('stock.stock_location_stock').id,
+             'location_id': operation.location_id.id or
+                self.env.ref('stock.stock_location_stock').id,
```
Comment (LoisRForgeFlow) [inline rma_purchase/wizards/rma_add_purchase.py:79]: I don't think using demo data like this is a good idea...

File: rma_sale/wizards/rma_add_sale.py [hunk 5]
```diff
@@ -74,8 +74,8 @@ def _prepare_rma_line_from_sale_order_line(self, line):
             {'in_route_id': operation.in_route_id.id or route,
              'out_route_id': operation.out_route_id.id or route,
              'receipt_policy': operation.receipt_policy,
-             'location_id': operation.location_id.id or
-                            self.env.ref('stock.stock_location_stock').id,
+             'location_id': operation.location_id.id or self.env.ref(
+                 'stock.stock_location_stock').id,
```
Comment (LoisRForgeFlow) [inline rma_sale/wizards/rma_add_sale.py]: same

**Review by LoisRForgeFlow — COMMENTED** [review]
to use demo data in logic as kind of 'default', are you sure that's a good option?

**Comment by LoisRForgeFlow** [comment]
My last commit (1e5995c006a0156a979b05c9ac07abfa8421db38) attend https://github.com/Eficent/stock-rma/issues/7

**Comment by LoisRForgeFlow** [comment]
I will merge this to continue working without messing up things until @aheficent is available to answer my comments

File: rma/models/product.py (hunk 3 above)
Comment (LoisRForgeFlow) [reply rma/models/product.py:11]: fixed in 1e5995c006a0156a979b05c9ac07abfa8421db38

File: rma_purchase/wizards/rma_add_purchase.py (hunk 4 above)
Comment (AaronHForgeFlow) [reply rma_purchase/wizards/rma_add_purchase.py:79]: I agree, remove that

**Pull request merged**